base/intfuncs.jl

```diff
@@ -200,6 +200,38 @@ end
 # However, we still need a fallback that calls the general ^:
 ^{p}(x, ::Type{Val{p}}) = x^p
 
+# for numbers, inline the power_by_squaring method for literal p.
+# (this also allows integer^-p to give a floating-point result
+#  in a type-stable fashion).
+@generated function ^{p}(x::Number, ::Type{Val{p}})
```

Comment:
`BigInt` overrides this.  We are already using `power_by_squaring` for `Complex{Float}` and in all other cases where this method is called, so there will be no accuracy regressions. "Inaccurate" seems like an exaggeration, since the roundoff errors grow extremely slowly, as `O(sqrt(log(p)))`.

Can you give an example of where this would cause a runtime performance issue compared to calling `power_by_squaring`?